### ISSUE TITLE
High memory traffic in nuget due to lots of NuGetVersion objects

### ISSUE THREAD

**Issue by cdmihai** [NuGet/Home#7291]
I was profiling memory traffic in MSBuild (single proc build of [WebLargeCore ](https://github.com/mikeharder/dotnet-cli-perf/tree/master/scenarios/web/large/core)with minimal logging) and noticed high memory traffic from nuget:

![image](https://user-images.githubusercontent.com/2255729/45324268-7efef400-b502-11e8-9daf-6fa9c5f68a30.png)
![image](https://user-images.githubusercontent.com/2255729/45324301-976f0e80-b502-11e8-8e62-9f2fe665ab7b.png)
![image](https://user-images.githubusercontent.com/2255729/45324324-a6ee5780-b502-11e8-8607-2624f25d5afd.png)

**Comment by rrelyea** [comment]
Potential perf improvement?

**Comment by cdmihai** [comment]
Yes, the fewer the allocations, the less GC is called. I'm being told that GC is quite the problem in VS.
